src/kani-compiler/cbmc/src/goto_program/location.rs

```diff
@@ -13,7 +13,14 @@ pub enum Location {
     BuiltinFunction { function_name: InternedString, line: Option<u64> },
     /// Location in user code.
     /// `function` is `None` for global, `Some(function_name)` for function local.
-    Loc { file: InternedString, function: Option<InternedString>, line: u64, col: Option<u64> },
+    Loc {
+        file: InternedString,
+        function: Option<InternedString>,
+        line: u64,
+        col: Option<u64>,
+        comment: InternedString,
+        property_class: InternedString,
```

Comment:
Just because CBMC 

1. Puts this on the location and 
2. Uses a string, doesn't mean we need to do either. Make this a propert of the `assert`, and use an `enum` for the different property classes. Then, make the `IRep` generator translate that into the anotations.

Reply:
Addressing both comments here - I think comment is a field that's common to statements, expressions as well but property_classes is not something that can be added to other things apart from `assert`. I'll change the flow to add these as properties to `assert` only. I was unclear about this earlier.

Reply:
@danielsn I though this package was trying to match CBMC's internal structure though.

That said, these new fields should be optional.